### PR TITLE
[SPARK-25662][SQL][TEST] Refactor DataSourceReadBenchmark to use main method

### PR DESCRIPTION
**Pull request by peter-toth**
## What changes were proposed in this pull request?

1. Refactor DataSourceReadBenchmark

## How was this patch tested?

Manually tested and regenerated results.
```
SPARK_GENERATE_BENCHMARK_FILES=1 build/sbt "sql/test:runMain org.apache.spark.sql.execution.benchmark.DataSourceReadBenchmark"
```

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DataSourceReadBenchmark.scala [hunk 1]
```diff
@@ -34,10 +34,15 @@ import org.apache.spark.sql.vectorized.ColumnVector
 
 /**
  * Benchmark to measure data source read performance.
- * To run this:
- *  spark-submit --class <this class> <spark sql test jar>
+ * To run this benchmark:
+ * {{{
+ *   1. without sbt: bin/spark-submit --class <this class> <spark sql test jar>
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DataSourceReadBenchmark.scala]: Hi, @peter-toth . Could you run this command actually?

Comment (peter-toth) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DataSourceReadBenchmark.scala]:
```
bin/spark-submit --class org.apache.spark.sql.execution.benchmark.DataSourceReadBenchmark --jars core/target/spark-core_2.11-3.0.0-SNAPSHOT-tests.jar,sql/catalyst/target/spark-catalyst_2.11-3.0.0-SNAPSHOT-tests.jar sql/core/target/spark-sql_2.11-3.0.0-SNAPSHOT-tests.jar
```
does work for me, but I checked in `FilterPushdownBenchmark` and it seems we don't mention other required jars.
Shall I modify the command?

Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/DataSourceReadBenchmark.scala]: Yes. We noticed that the required jars are introduced during refactoring. So, we start to fix this guide in recent PRs like [this](https://github.com/apache/spark/commit/669ade3a8eed0016b5ece57d776cea0616417088).

**Comment by dongjoon-hyun** [comment]
ok to test

**Comment by dongjoon-hyun** [comment]
Hi, @peter-toth .
Could you review and merge https://github.com/peter-toth/spark/pull/1 which contains the result on EC2 r3.xlarge?

**Comment by dongjoon-hyun** [comment]
Could you add `[SQL]` before `[TEST]`, too?

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Comment by dongjoon-hyun** [comment]
Hi, @dbtsai .
Since this is `DataSourceReadBenchmark`, could you review and merge this?

**Comment by dongjoon-hyun** [comment]
Retest this please.

**Comment by dbtsai** [comment]
Thanks @dongjoon-hyun for ping me. LGTM too. We're working on some parquet reader improvement, and this will be useful.

**Comment by dongjoon-hyun** [comment]
Thank you, @dbtsai and @peter-toth .

**Comment by dbtsai** [comment]
@peter-toth I assigned to you. Thanks for contribution.

**Pull request closed without merging**